src/main/tray.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0
+ * Vesktop, a desktop app aiming to give you a snappier Discord Experience
+ * Copyright (c) 2023 Vendicated and Vencord contributors
+ */
+
+import { dialog, NativeImage, nativeImage } from "electron";
+import { mkdirSync, readFileSync, writeFileSync } from "fs";
+import { readFile } from "fs/promises";
+import { join } from "path";
+import { IpcEvents } from "shared/IpcEvents";
+import { BADGE_DIR, ICONS_DIR, STATIC_DIR } from "shared/paths";
+
+import { lastBadgeCount } from "./appBadge";
+import { mainWin, tray } from "./mainWindow";
+import { Settings } from "./settings";
+
+export const statusToSettingsKey = {
+    speaking: "traySpeakingOverride",
+    muted: "trayMutedOverride",
+    deafened: "trayDeafenedOverride",
+    idle: "trayIdleOverride",
+    icon: "trayMainOverride"
+};
+
+export const isCustomIcon = (status: string) => {
+    const settingKey = statusToSettingsKey[status as keyof typeof statusToSettingsKey];
+    return Settings.store[settingKey];
+};
+
+export async function setTrayIcon(iconName: string) {
+    if (!tray || tray.isDestroyed()) return;
+    const Icons = new Set(["speaking", "muted", "deafened", "idle", "icon"]);
+    if (!Icons.has(iconName)) return;
+
+    // if need to set main icon then check whether there is need of notif badge
+    if (iconName === "icon" && lastBadgeCount > 0) {
```

Comment:
I'm not sure if I understand the new behavior correctly. After updating, it now shows the dot for non-mentions again.
Am I supposed to disable “Enable Unread Message Badge” when I want it to only for mentions?
![image](https://github.com/Vencord/Vesktop/assets/24937357/502ad68b-0583-463f-be67-db42300ab810)

I think this could be made clearer